qemu/tests/balloon_hotplug.py

```diff
@@ -51,6 +69,13 @@ def run(test, params, env):
                               logging.info)
         balloon_check.run(test, params, env)
 
+        if pm_test_after_plug:
+            run_pm_test(pm_test_after_plug, "hot-plug")
+            if vm.is_alive():
+                balloon_check.run(test, params, env)
+            else:
```

Comment:
What do you want to achieve here?
if `vm` is not alive, do nothing?

Reply:
pm_test_after_plug mainly include reboot and shutdown.
We need to check if device works well after guest reboot,but it's not necessary if guest is shutdown.
So I will check if guest is alive, if no pm_test_after_plug is shutdown,skip followed tests.
I considered use if pm_test_after_plug == "shutdown": return.But shuang suggest it's not good to filter out a specific case in .py file.

Reply:
I will use continue instead of return

Reply:
OK. Please make a comment in the code, all it maybe difficult for people to understand

Reply:
It seems for `pm_test_after_plug == "shutdown"`,  `balloon_repeats` must be 1

Reply:
Modified,add a line to explain it.
And yes,if run shutdown subtest,the repeat time must be 1,1 is the default value.
Thank you